src/ui/console.c

```diff
@@ -862,6 +866,49 @@ cons_show_disco_contact_information(GHashTable* addresses)
     }
 }
 
+void
+cons_show_omemo_qrcode(const char* const text)
+{
+#ifdef HAVE_QRENCODE
+    static const size_t ZOOM_SIZE = 10;
+    QRcode* qrcode = QRcode_encodeString(text, 0, QR_ECLEVEL_L, QR_MODE_8, 1);
+
+    int width = (qrcode->width * ZOOM_SIZE);
+    unsigned char* data = qrcode->data;
+
+    ProfWin* console = wins_get_console();
+
+    char buf[(width * 4) + 1];
+    memset(buf, 0, sizeof buf);
+
+    char tmp[(width * 4) + 5];
+    memset(tmp, 0, sizeof tmp);
```

Comment:
Please don't use VLA's. Usage has been abandoned in most projects for good reasons.

We should maybe add a `-Werror=vla` so this is found while developing stuff.

Reply:
```suggestion
    char *buf = calloc((width * 4) + 1, 1);
    char *tmp = calloc((width * 4) + 5, 1);
    if (!buf || !tmp) {
        free(tmp);
        free(buf);
        return;
    }
```

and before returning at the end, also `free()` the two buffers.

Reply:
Oh yeah I forgot calloc exists. Thank you.

When I find the actual format for an OMEMO QR code, I'll go ahead and implement this change in a separate PR.

Reply:
Already done in https://github.com/profanity-im/profanity/pull/1718 :)

Reply:
Sweet, less work for me:)

I will make the change when I'm home, which should be Sunday night